storage/src/main/java/tech/pegasys/teku/storage/client/CombinedChainDataClient.java

```diff
@@ -421,6 +421,21 @@ private boolean isRecentData(final UInt64 slot) {
     return Optional.ofNullable(getStore().getLatestFinalizedBlockAndState().getBlock());
   }
 
+  public Optional<BeaconState> getFinalizedState() {
+    if (recentChainData.isPreGenesis()) {
+      return Optional.empty();
+    }
+
+    return Optional.ofNullable(getStore().getLatestFinalizedBlockAndState().getState());
```

Comment:
nit: I think this should just be
```suggestion
    return Optional.of(getStore().getLatestFinalizedBlockAndState().getState());
```

because if we have a store it should have a finalized state.